app/src/androidTest/java/com/github/se/eventradar/viewProfile/ProfileTest.kt

```diff
@@ -0,0 +1,196 @@
+  @get:Rule val composeTestRule = createComposeRule()
+  // This rule automatic initializes lateinit properties with @MockK, @RelaxedMockK, etc.
+  @get:Rule val mockkRule = MockKRule(this)
+  // Relaxed mocks methods have a default implementation returning values
+  @RelaxedMockK lateinit var mockNavActions: NavigationActions
+
+  @RelaxedMockK lateinit var mockProfileViewModel: ProfileViewModel
+  private val mockUser =
+      User(
+          userId = "1",
+          birthDate = "01/01/2000",
+          email = "test@example.com",
+          firstName = "John",
+          lastName = "Doe",
+          phoneNumber = "1234567890",
+          accountStatus = "active",
+          eventsAttendeeList = mutableListOf("event1", "event2"),
+          eventsHostList = mutableListOf("event3"),
+          friendsList = mutableListOf("2"),
+          profilePicUrl = "http://example.com/Profile_Pictures/1",
+          qrCodeUrl = "http://example.com/QR_Codes/1",
+          bio = "",
+          username = "johndoe")
+  private val mockFriend =
+      User(
+          userId = "2",
+          birthDate = "02/02/2002",
+          email = "friend@example.com",
+          firstName = "Jim",
+          lastName = "Smith",
+          phoneNumber = "1234567890",
+          accountStatus = "active",
+          eventsAttendeeList = mutableListOf("event1", "event2"),
+          eventsHostList = mutableListOf("event3"),
+          friendsList = mutableListOf("1"),
+          profilePicUrl = "http://example.com/Profile_Pictures/2",
+          qrCodeUrl = "http://example.com/QR_Codes/2",
+          bio = "",
+          username = "jimsmith")
+  private val sampleUiState =
+      MutableStateFlow(
+          ProfileUiState(
+              profilePicUrl = "",
+              firstName = "Jim",
+              username = "Smith",
+              bio = "I am Jim Smith and I love the Smiths (the band)."))
+
+  @Before
+  fun testSetup() {
+    every { mockProfileViewModel.getProfileDetails() } returns Unit
+    every { mockProfileViewModel.uiState } returns sampleUiState
+  }
+
+  @Test
+  fun screenDisplaysAllElementsCorrectlyWhenPublic() = run {
+    composeTestRule.setContent {
+      ProfileUi(
+          isPublicView = true, viewModel = mockProfileViewModel, navigationActions = mockNavActions)
+    }
+
+    ComposeScreen.onComposeScreen<ProfileScreen>(composeTestRule) {
+      chatButton { assertIsDisplayed() }
+      goBackButton { assertIsDisplayed() }
+      bottomNav { assertIsDisplayed() }
+      centeredViewProfileColumn { assertIsDisplayed() }
+      profilePic { assertIsDisplayed() }
+      name { assertIsDisplayed() }
+      username { assertIsDisplayed() }
+      leftAlignedViewProfileColumn { assertIsDisplayed() }
+      bioLabelText { assertIsDisplayed() }
+      bioInfoText { assertIsDisplayed() }
+      // In public view, phone number and birth date should not be displayed
```

Comment:
I agree. With the current development, the separation between Private and Public profiles is not very valuable. However, this does make sense for future development and the phone number and birth date fields are the less legitimate public information (it can even be argued that you never want your phone number to be shared for example)